src/tests/suite/daos_test_common.c

```diff
@@ -1017,3 +1019,106 @@ get_daos_prop_with_user_acl_perms(uint64_t perms)
 	D_FREE(user);
 	return prop;
 }
+
+/* JSON output handling for dmg command */
+static struct json_object *daos_dmg_json_contents(const char *dmg_cmd)
+{
+	long int  size = 0;
+	char *content = NULL;
+	char *filename = "/tmp/daos_dmg.json";
+	struct json_object *parsed_json;
+	int	  rc;
+
+	FILE* fp = fopen(filename, "w+");
+	if (!fp) {
+		print_message("fopen %s failed!\n", filename);
+		return NULL;
+	}
+
+	/* Need to make sure -j is in the dmg cmd? */
+	rc = system(dmg_cmd);
+	assert_int_equal(rc, 0);
+
+	/* get the content size and allocate buffer */
+	fseek(fp, 0, SEEK_END);
+	size = ftell(fp);
+	// rewind(fp);
+	fseek(fp, 0, SEEK_SET);
+	D_ALLOC(content, size);
+	print_message("json output size is %ld\n", size);
+
+        if ( fread(content, size, 1, fp) != 1 ) {
```

Comment:
(style)  code indent should use tabs where possible
(style)  space prohibited after that open parenthesis '('
(style)  space prohibited before that close parenthesis ')'